iwyu_ast_util.cc

```diff
@@ -546,32 +560,29 @@ class TypeEnumerator : public RecursiveASTVisitor<TypeEnumerator> {
 
   // --- Methods on RecursiveASTVisitor
   bool TraverseType(QualType type) {
-    TraverseArgumentsOfSugaredTemplates(type);
-    return Base::TraverseType(type);
+    return TraverseTypeHelper(type);
   }
 
   bool TraverseTypeLoc(TypeLoc type_loc) {
-    TraverseArgumentsOfSugaredTemplates(type_loc.getType());
-    return Base::TraverseTypeLoc(type_loc);
-  }
-
-  bool VisitType(Type* type) {
-    seen_types_.insert(type);
-    return true;
+    return TraverseTypeHelper(type_loc.getType());
   }
 
  private:
-  // Clang doesn't traverse underlying type for most of the sugar types.
-  // If a TemplateSpecializationType occurs in a sugaring chain, traverse its
-  // arguments explicitly, otherwise they aren't put into resugar_map.
-  void TraverseArgumentsOfSugaredTemplates(QualType type) {
-    if (type.isNull())
-      return;
-
-    if (const auto* template_spec = type->getAs<TemplateSpecializationType>()) {
-      for (const TemplateArgument& arg : template_spec->template_arguments())
-        TraverseTemplateArgument(arg);
-    }
+  bool TraverseTypeHelper(QualType qual_type) {
+    if (qual_type.isNull())
+      return false;
+
+    const Type* const type = qual_type.getTypePtr();
```

Comment:
No need for the second `const` in `const Type* const`. I prefer to only use that when forced by the standard library :). Shorter lines with fewer tokens are easier to read.